src/components/button/index.jsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable react/button-has-type */
+/*
+	Disabled ESLint rule as the type attribute is using oneOf propType with a default
+	value - whihch ESLint is unable to parse as acceptable when in the use case below
+	is valid as there is a default and the use of oneOf PropType validation
+*/
+import { forwardRef } from "react";
+import PropTypes from "prop-types";
+import classnames from "classnames";
+
+const COMPONENT_CLASS_NAME = "c-button";
+
+const Button = forwardRef((props, ref) => {
+	const {
+		accessibilityLabel,
+		children,
+		className,
+		disabled,
+		fullWidth,
+		href,
+		iconLeft,
+		iconRight,
+		size,
+		style,
+		type,
+		...rest
+	} = props;
+
+	const defaultProps = {
+		...rest,
+		ref,
+		"aria-label": accessibilityLabel,
+		className: classnames({
```

Comment:
not totally sure I understand how this lib is better than string interpolation. I had to lookup how this library works to understand these lines of code

Reply:
With the amount of classNames we can have with the button, doing string interpolation looked very messy - essential with this library if the value is a truthy value it would add the key value to the class list

Reply:
@JackHowa - How would you accomplish it without a library?

Reply:
as long as we are bringing this in, it could also be used for the "classname merge" that we have in all the components.  It's cleaner than the "append" method we have currently.

Reply:
I get that interpolation can look messy. In terms of readability, and for new or newer devs, this seems more readable to me compared with the current text. 

One suggestion that would help me as reader is a comment above classname that might say what the library does. That might be helpful if I was looking at this with fresh eyes. 

```jsx
	const defaultProps = {
		...rest,
		ref,
		"aria-label": accessibilityLabel,
		className: `
			${COMPONENT_CLASS_NAME}
			${href ? ` ${COMPONENT_CLASS_NAME}--link` : ""}
			${fullWidth ? ` ${COMPONENT_CLASS_NAME}--full-width` : ""}
			${size ? ` ${COMPONENT_CLASS_NAME}--${size}` : ""}
			${style ? ` ${COMPONENT_CLASS_NAME}--${style}` : ""}
			${className ? ` ${className}` : ""}
		`,
	};
```

Reply:
With Prettier installed that translates too

```
		className: `${COMPONENT_CLASS_NAME}${href ? ` ${COMPONENT_CLASS_NAME}--link` : ""}${
			fullWidth ? ` ${COMPONENT_CLASS_NAME}--full-width` : ""
		}${size ? ` ${COMPONENT_CLASS_NAME}--${size}` : ""}${
			style ? ` ${COMPONENT_CLASS_NAME}--${style}` : ""
		}${className ? ` ${className}` : ""}`,
```

which is very hard to follow what's going on

Reply:
IMO, the `classnames` library is easier to read and I don't think it's too hard to see what's going on for someone new to it.

If you format it as this, prettier won't touch it:
```
		className: `
			${COMPONENT_CLASS_NAME}
			${href ? ` ${COMPONENT_CLASS_NAME}--link` : ""}
			${fullWidth ? ` ${COMPONENT_CLASS_NAME}--full-width` : ""}
			${size ? ` ${COMPONENT_CLASS_NAME}--${size}` : ""}
			${style ? ` ${COMPONENT_CLASS_NAME}--${style}` : ""}
			${className ? ` ${className}` : ""}
		`,
```
...but it would also linebreak the classname property in the rendered output.

There is an alternative if we have an issue with the popular `classnames` that offers some additional features: https://www.npmjs.com/package/classd  (but I think `classnames` is probably fine)

I do see a lot of disfavor on-line toward using `classnames` though and I don't know why.

Reply:
yeah I'm fine if this is not the way we go. one takeaway for me is that a comment above the lib would have been helpful for me to understand what was happening. Is that an ok compromise? @matthewroach 

And explicit, albeit messy (with prettier), would be preferable. As a bias, I care less if something is 'clean' or code-golf shortest solution. Also, when I see a ternary, I know it's a bool check. Whereas, in that lib, I didn't know that the second arg was a bool check

Reply:
@JackHowa `classnames` can also be used like this since falsey values are ignored:
```
		className: classnames(
			COMPONENT_CLASS_NAME,
			href && `${COMPONENT_CLASS_NAME}--link`,
			fullWidth && `${COMPONENT_CLASS_NAME}--full-width`,
			size && `${COMPONENT_CLASS_NAME}--${size}`,
			style && `${COMPONENT_CLASS_NAME}--${style}`,
			className,
		),
```

Reply:
also ran across this library that claims to be faster: https://www.npmjs.com/package/clsx

Reply:
@nschubach : Great find! Here's an Npm Trends comparison:
[npmtrends.com/clsx-vs-classnames](https://www.npmjs.com/package/clsx)

The size difference is marginal but the alleged performance gain is intriguing!
https://bundlephobia.com/package/classnames@2.3.1
https://bundlephobia.com/package/clsx@1.1.1

The readme for `clsx` shows benchmarks between the two:
https://github.com/lukeed/clsx/tree/master/bench#benchmarks

Reply:
I looked at clsx but their document was very limited - and I wasn't sure if they followed the `classnames` functionality - if we are using a third party library I want the documentation to be reliable

Reply:
> I am not sold on the comment above the library - as the more we use the library do we still add a comment every-time we use it? Feels like an overly cautious approach that I don't fully understand.

Against adding one comment in multiple places? Adding comments to me helps comprension wherever I see something confusing, even more so if repeated. 

If opposed to a one-line comment "// the value of the key-value pair is a boolean check to determine whether or not to show the key as a class"?, is there another way we can be explicit? For example, what about explicitly cast the values as boolean because that's how they're being used:

```
[`${COMPONENT_CLASS_NAME}--full-width`]: Boolean(fullWidth),
```

Overall, my initial impression to this code was surprise that we were both passing a boolean as well as a string as a value into a library I had never seen before. In my mind, this is a minor readability suggestion. I haven't looked at other libraries doing conditional string interpolation.

Reply:
Update the implementation to remove the need for the `classnames` library - having looked at @nschubach example about it occurred to me we can do the same logic with a basic array